feat: Add docs page for Remix SPA Mode (#1208)
clerk/clerk-docs: docs/references/remix/spa-mode.mdx

---
title: SPA Mode
description: Clerk supports Remix SPA mode out-of-the-box.
---

# SPA Mode

> After implementing the [Remix SPA Mode changes](https://remix.run/docs/en/main/guides/spa-mode), you need to make the following changes to your Remix app to use Clerk in SPA mode.
> 
> 1. Remove the `rootAuthLoader` from your `app/root.tsx` file.
> 2. Pass the Publishable Key and any other Environment Variables to the ClerkApp component (except the Secret Key, which should not be used on the client side).
> 3. Remove all usages of server-side Clerk functions (e.g. all the imported functions coming from `@clerk/remix/ssr.server`)

Reviewer: ```suggestion
After implementing the [Remix SPA Mode changes](https://remix.run/docs/en/main/guides/spa-mode), you need to make the following changes to your Remix app to use Clerk in SPA mode:

1. Remove the `rootAuthLoader()` from your `app/root.tsx` file.
2. Pass the publishable key and any other environment variables to the [`ClerkApp`](/docs/references/remix/clerk-app) component (except the secret key, which should never be used on the client side).
3. Remove all usages of server-side Clerk functions (for example, all the imported functions coming from `@clerk/remix/ssr.server`)
```

> ## Example root page

Reviewer: ```suggestion
## Example `root.tsx` page
```

> An example `app/root.tsx` page that uses SPA Mode, could be the following:

Reviewer: ```suggestion
The following example demonstrates an `app/root.tsx` file that uses SPA mode. Notice the absence of the `rootAuthLoader()`.
```

> ```tsx

Reviewer: ```suggestion
```tsx {{ filename: 'app/root.tsx' }}
```

import { ClerkApp } from "@clerk/remix";
import {
  Links,
  Meta,
  Outlet,
  Scripts,
  ScrollRestoration,
} from "@remix-run/react";

function App() {
  return (
    <html lang="en">
      <head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <Scripts />
      </body>
    </html>
  );
}

export default ClerkApp(App, {
  publishableKey: 'pk_test_XYZ',
  signInUrl: '/sign-in',
  signUpUrl: '/sign-up',
  signInFallbackRedirectUrl: '/',
  signUpFallbackRedirectUrl: '/',
});
```

## Example protected route

> An example of a protected route that uses SPA Mode, could be the following:

Reviewer: ```suggestion
The following example demonstrates how to use Clerk's [control components](/docs/components/overview) in order to protect a route that uses SPA mode. 
```
